### PR TITLE
add ring/kton burndrop contract

### PR DESCRIPTION
**Pull request by WoeOm**

### CODE REVIEW

File: contracts/TokenBurnDrop.sol [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    * @param _amount - amount of token.
+    * @param _data - data which indicate the operations.
+    */
+    function tokenFallback(address _from, uint256 _amount, bytes _data) public {
+        address ring = registry.addressOf(SettingIds.CONTRACT_RING_ERC20_TOKEN);
+        address kryptonite = registry.addressOf(SettingIds.CONTRACT_KTON_ERC20_TOKEN);
+
+        require((msg.sender == ring) || (msg.sender == kryptonite), "Permission denied");
+
+        require(_data.length == 32, "The address (Darwinia Network) must be in a 64-bit hexadecimal format");
```
Comment (HackFisher) [inline contracts/TokenBurnDrop.sol]: Add more check to for the receiver subkey address.

File: contracts/TokenBurnDrop.sol [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        if(ring == msg.sender) {
+            ERC223(ring).transfer(address(0), _amount);
+        }
+
+        //  burndrop kton
+        if (kryptonite == msg.sender) {
+            ERC223(kryptonite).transfer(address(0), _amount);
+        }
+
+        emit BurndropTokens(msg.sender, _from, _amount, _data);
```
Comment (HackFisher) [inline contracts/TokenBurnDrop.sol]: If the msg.sender if not from RING or KTON, this event will also be triggers, which will the verifiers' effort to filter and check the sender.

Better to filter in the contract here.

File: contracts/TokenBurnDrop.sol (hunk 1 above)
Comment (WoeOm) [reply contracts/TokenBurnDrop.sol]:
```suggestion
        require(_data.length == 33, "The address (Darwinia Network) must be in a 33 bytes hexadecimal format");
        require(byte(_data[0]) == SS58_PREFIX_DARWINIA, "Darwinia Network Address ss58 prefix is 42");
        require(darwiniaAddress != bytes32(0x0), "Darwinia Network Address can't be empty");
```

File: contracts/TokenBurnDrop.sol (hunk 2 above)
Comment (WoeOm) [reply contracts/TokenBurnDrop.sol]:
```suggestion
        require((msg.sender == ring) || (msg.sender == kryptonite), "Permission denied");
        emit BurndropTokens(msg.sender, _from, _amount, _data);
```

**Pull request merged**